homeassistant/helpers/entity_platform.py

```diff
@@ -418,7 +418,7 @@ async def async_add_entities(
         This method must be run in the event loop.
         """
         # handle empty list from component/platform
-        if not new_entities:
+        if not new_entities:  # type: ignore[truthy-iterable]
             return
```

Comment:
I considered changing `new_entities` to `Collections[Entity]`. As it turns out though, quite a few integrations actually call it with a `Generator`. Those would need to be changed first.

In the end, it's probably not worth it here. This is an early out anyway. All cases would also be caught a few lines below with
https://github.com/home-assistant/core/blob/4fbb14ecc7135bf7c2adf6f71d3ffdcc79013e1a/homeassistant/helpers/entity_platform.py#L428-L437